### PR TITLE
fix writternBy delete by optimizer

### PR DESCRIPTION
**Pull request by heroicNeZha**
#### What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

#### What problem(s) does this PR solve?
Issue(s) number: 
closing #3652 
Description:
```
# before fixed 
(root@nebula) [nba]> match (v:player{name:"Tony Parker"}) with v as a match p=(o:player{name:"Tim Duncan"})-[]->(a) return o.player.name
+---------------+
| o.player.name |
+---------------+
+---------------+
Empty set (time spent 2825/3306 us)

Thu, 06 Jan 2022 16:43:39 CST

# after fixed
(root@nebula) [nba]> match (v:player{name:"Tony Parker"}) with v as a match p=(o:player{name:"Tim Duncan"})-[]->(a) return o.player.name
+---------------+
| o.player.name |
+---------------+
| "Tim Duncan"  |
| "Tim Duncan"  |
+---------------+
Got 2 rows (time spent 9883/24329 us)

Tue, 18 Jan 2022 15:40:35 CST
```

image

There is a `writternBy` attribute of `__project_5` in symbolTable , but it was deleted when the optimizer released
So `ArgumentExecutor` will not build futureMap of `Arugument_6` nodes

#### Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

#### Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Review by CPWstatic — APPROVED** [review]
Excellent job!!!

File: src/graph/planner/plan/PlanNode.cpp [hunk 1]
```diff
@@ -370,6 +370,13 @@ void PlanNode::releaseSymbols() {
   }
 }
 
+void PlanNode::updateSymbols() {
+  auto symTbl = qctx_->symTable();
+  for (auto out : outputVars_) {
+    out && symTbl->updateWrittenBy(out->name, out->name, this);
```
Comment (Shylock-Hg) [inline src/graph/planner/plan/PlanNode.cpp:376]: if (out != nullptr)

Comment (heroicNeZha) [reply src/graph/planner/plan/PlanNode.cpp:376]: if out is nullptr, `out && updateWrittenBy` updateWrittenBy will be cut off

Comment (Shylock-Hg) [reply src/graph/planner/plan/PlanNode.cpp:376]: It's not a clear code style.

Comment (heroicNeZha) [reply src/graph/planner/plan/PlanNode.cpp]: thanks, fixed

**Review by Shylock-Hg — APPROVED** [review]

**Review by CPWstatic — APPROVED** [review]

**Pull request merged**